SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -0,0 +1,226 @@
+		const ANSICHAR* Value = ValueSrc.Get();
+		Trace_EventData_AddStringFields(EventData, 1, &Key, &Value);
+	};
+
+#if ENGINE_MINOR_VERSION >= 25
+	typedef FProperty UnrealProperty;
+	typedef FStrProperty UnrealStrProperty;
+	typedef FObjectProperty UnrealObjectProperty;
+#else
+	typedef UProperty UnrealProperty;
+	typedef UStrProperty UnrealStrProperty;
+	typedef UObjectProperty UnrealObjectProperty;
+#endif
+	for (TFieldIterator<UnrealProperty> It(Struct); It; ++It)
+	{
+		UnrealProperty* Property = *It;
+
+		FString VariableName = Property->GetName();
+		const void* Value = Property->ContainerPtrToValuePtr<uint8>(&EventMessage);
+
+		check(Property->ArrayDim == 1); // Arrays not handled yet
+
+		// convert the property to a FJsonValue
+		if (UnrealStrProperty *StringProperty = CastField<UnrealStrProperty>(Property))
+		{
+			AddTraceEventStringField(VariableName, StringProperty->GetPropertyValue(Value));
+		}
+		else if (UnrealObjectProperty* ObjectProperty = CastField<UnrealObjectProperty>(Property))
+		{
+			UObject* Object = ObjectProperty->GetPropertyValue(Value);
+			if (Object)
+			{
+				AddTraceEventStringField(VariableName, Object->GetName());
+				if (AActor* Actor = Cast<AActor>(Object))
+				{
+					FString KeyString = VariableName + TEXT("Position");
+					FString ValueString = Actor->GetTransform().GetTranslation().ToString();
+					AddTraceEventStringField(KeyString, ValueString);
+				}
+			}
+			else
+			{
+				AddTraceEventStringField(VariableName, "Null");
+			}
+		}
+		else // Default
+		{
+			FString StringValue;
+			Property->ExportTextItem(StringValue, Value, NULL, NULL, PPF_None);
+			AddTraceEventStringField(VariableName, StringValue);
+		}
+
+		// else if (UEnumProperty* EnumProperty = Cast<UEnumProperty>(Property))
+		// else if (UNumericProperty *NumericProperty = Cast<UNumericProperty>(Property))
+		// else if (UBoolProperty *BoolProperty = Cast<UBoolProperty>(Property))
+		// else if (UTextProperty *TextProperty = Cast<UTextProperty>(Property))
+		// else if (UArrayProperty *ArrayProperty = Cast<UArrayProperty>(Property))
+		// else if (USetProperty* SetProperty = Cast<USetProperty>(Property))
+		// else if (UMapProperty* MapProperty = Cast<UMapProperty>(Property))
+		// else if (UStructProperty *StructProperty = Cast<UStructProperty>(Property))
+
+	}
+
+	// TODO(EventTracer): implement and call AddTargetObjectInfoToEventData
+	// TODO(EventTracer): implement and call AddComponentIdInfoToEventData
+
+	TraceEvent.data = EventData;
+	Trace_EventTracer_AddEvent(EventTracer, &TraceEvent);
+	Trace_EventData_Destroy(EventData);
+
+	return CurrentSpanId;
+}
+
+void SpatialEventTracer::Enable(const FString& FileName)
+{
+	Trace_EventTracer_Parameters parameters = {};
+	parameters.user_data = this;
+	parameters.callback = &SpatialEventTracer::TraceCallback;
+	EventTracer = Trace_EventTracer_Create(&parameters);
+	Trace_EventTracer_Enable(EventTracer);
+
+	bEnabled = true;
```

Comment:
Get rid of this and instead just use `Trace_EventTracer_IsEnabled()` if you need to?

Reply:
Trace_EventTracer_IsEnabled is not available to us currently. Perhaps we need a newer version of the library?